### PR TITLE
Test coverage and refactor

### PR DESCRIPTION
**Pull request by ziobron**
Please take a look and comment.
If it's ok please Approve.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -1,14 +1,29 @@
 #include <iostream>
+#include <tuple>
```
Comment (ziobron) [inline src/main.cpp:2]: Not needed. To be removed.

File: tests/UserHandlerTests.cpp [hunk 2]
```diff
@@ -53,9 +65,20 @@ TEST_F(UserHandlerTests, updateUserAffectRequestedUser)
...
+}
+
+TEST_F(UserHandlerTests, clearUserDatabaseErasesWholeDb)
+{
+    cut.addUser(szymon);
+    cut.addUser(shrek);
+    auto beforeClear = cut.getUserDatabase();
+    cut.clearUserDatabase();
+    auto afterClear = cut.getUserDatabase();
+    ASSERT_NE(beforeClear, afterClear);
```
Comment (adamvm) [inline tests/UserHandlerTests.cpp:82]: For me this condition is not so precise, I prefer explicit values asserting

**Review by adamvm — APPROVED** [review]
Fine for me, no errors found

File: src/CourseDashboard.cpp [hunk 3]
```diff
@@ -2,31 +2,62 @@
 #include "FileHandler.hpp"
 #include "Utility.hpp"
 #include "AuthenticationProvider.hpp"
+#include <iostream>
 
 
-void CourseDashboard::loadFromFile(const std::string& pathTofile)
+void CourseDashboard::loadFromJson(const std::string& pathToFile)
```
Comment (adamvm) [inline src/CourseDashboard.cpp:8]: I would keep loadFromFile and later if required use design pattern (e.g. factory for different format file readers) by user

**Review by adamvm — APPROVED** [review]
Fine for me, no errors found

**Review by adamvm — APPROVED** [review]
Fine for me, no errors found

File: tests/AuthenticationProviderTests.cpp [hunk 4]
```diff
@@ -4,68 +4,37 @@
...
+TEST_F(AuthenticationProviderTests, canNotAuthenticateWrongEmail)
 {
-    User user("Alicja", "AJU", User::Group::evening, "alicjaliQui", "alicjaliQui", "alicjaliqui@gmail.com", "");
-    AuthenticationProvider authenticator;
-
-    std::string email = "alicjaliqui@gmail.com";
-    std::string password = "";
-
-    EXPECT_EQ(authenticator(user, email, password), true);
+    EXPECT_FALSE(authenticator.authenticate(user, "szymon@gmail.com", password));
```
Comment (adamvm) [inline tests/AuthenticationProviderTests.cpp:27]: Can generate some additional portion of spam for some Szymon ;) - open repo for everyone

**Review by adamvm — APPROVED** [review]
Fine for me, no errors found

**Comment by adamvm** [comment]
:+1:

**Pull request merged**